ipn/ipnlocal/autoupdate.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package ipnlocal
```

Comment:
do we need all this code on, say, iOS?

Reply:
We can also skip it on App Store macOS.

Reply:
Split the file with build constraints.
Note that these auto-updates won't start on nodes that don't set `AutoUpdate.Apply` in prefs, which can only be set on platforms that support auto-updates (so, not iOS/Android/non-standalone macOS/etc). So this is mostly about excluding additional compiled code in the final binary.

Reply:
We should also skip it on macsys. Sparkle already does its own automatic updates thing in the background regardless of whether the network extension is running. Admins can set up their own update check interval via MDM.

Reply:
Excluded `darwin` from the build constraints.
I just realized [we don't implement](https://github.com/tailscale/tailscale/blob/87e855342ca4f182186d00d160db94713f503f0b/clientupdate/clientupdate.go#L221-L233) auto-updates even for standalone macOS installs.